Mathlib/Algebra/Order/GroupWithZero/Canonical.lean

```diff
@@ -243,6 +243,43 @@ lemma pow_lt_pow_succ (ha : 1 < a) : a ^ n < a ^ n.succ := by
 lemma pow_lt_pow_right₀ (ha : 1 < a) (hmn : m < n) : a ^ m < a ^ n := by
   induction' hmn with n _ ih; exacts [pow_lt_pow_succ ha, lt_trans ih (pow_lt_pow_succ ha)]
 
+lemma zpow_lt_zpow_of_lt_one₀ (ha₀ : 0 < a) (ha₁ : a < 1) {m n : ℤ} (hmn : m < n) : a ^ n < a ^ m :=
+  zpow_right_strictAnti₀ ha₀ ha₁ hmn
+
+lemma le_of_zpow_le_zpow₀ (ha : 1 < a) {n m : ℤ} (hmn : a ^ m ≤ a ^ n) :
+    m ≤ n := by
+  contrapose! hmn
+  exact zpow_lt_zpow_right₀ ha hmn
+
+lemma le_of_zpow_le_zpow_of_lt_one₀ (ha₀ : 0 < a) (ha₁ : a < 1) {m n : ℤ}
+    (hmn : a ^ m ≤ a ^ n) :
+    n ≤ m := by
+  contrapose! hmn
+  exact zpow_lt_zpow_of_lt_one₀ ha₀ ha₁ hmn
+
+lemma eq_zero_of_zero_pow_eq_one₀ (ha : (0 : α) ^ n = 1) : n = 0 := by
+  induction n with
+  | zero => rfl
+  | succ n _ => simp at ha
+
+lemma eq_zero_of_zero_zpow_eq_one₀ {n : ℤ} (ha : (0 : α) ^ n = 1) : n = 0 := by
+  wlog hn : 0 ≤ n
+  · rw [← Int.neg_eq_zero]
+    apply this (α := α)
+    · simpa
+    · omega
+  have : n.toNat = 0 := eq_zero_of_zero_pow_eq_one₀ (α := α) <| by
+    rwa [← zpow_natCast, Int.toNat_of_nonneg hn]
+  omega
+
+lemma eq_zero_of_zpow_eq_one₀ (ha : a < 1) {n : ℤ} (han : a ^ n = 1) : n = 0 := by
```

Comment:
This can be made an iff, weakened to `a ≠ 1`, and should be proved using and moved close to `zpow_right_inj₀`

Reply:
done.